src/textual/css/_help_text.py

```diff
@@ -0,0 +1,690 @@
+from __future__ import annotations
+
+import sys
+from dataclasses import dataclass
+
+from textual.css._help_renderables import Example, Bullet, HelpText
+from textual.css.constants import (
+    VALID_BORDER,
+    VALID_LAYOUT,
+    VALID_EDGE,
+    VALID_ALIGN_HORIZONTAL,
+    VALID_ALIGN_VERTICAL,
+    VALID_STYLE_FLAGS,
+)
+
+if sys.version_info >= (3, 8):
+    from typing import Literal, Iterable
+else:
+    from typing_extensions import Literal
+
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    from typing_extensions import Annotated
+
+from textual.css._error_tools import friendly_list
+from textual.css.scalar import SYMBOL_UNIT
+
+StylingContext = Annotated[
+    Literal["inline", "css"],
+    """The type of styling the user was using when the error was encountered.
+Used to give help text specific to the context i.e. we give CSS help if the
+user hit an issue with their CSS, and Python help text when the user has an
+issue with inline styles.""",
+]
+
+
+@dataclass
+class ContextSpecificBullets:
+    """
+    Args:
+        inline (Iterable[Bullet]): Information only relevant to users who are using inline styling.
+        css (Iterable[Bullet]): Information only relevant to users who are using CSS.
+    """
+
+    inline: Iterable[Bullet]
```

Comment:
Some Iterables can be exhausted, so you might not get the same result from `get_by_context` on a second call.

Could these be Sequene or List ?